internal/server/handlers.go

```diff
@@ -199,7 +212,18 @@ func (a *API) Login(c *gin.Context, r *api.LoginRequest) (*api.LoginResponse, er
 			return nil, fmt.Errorf("update provider client: %w", err)
 		}
 
-		loginMethod = authn.NewOIDCAuthentication(r.OIDC.ProviderID, r.OIDC.RedirectURL, r.OIDC.Code, providerClient)
+		loginMethod, err = authn.NewOIDCAuthentication(
+			provider,
+			r.OIDC.RedirectURL,
+			r.OIDC.Code,
+			providerClient,
+			// TODO: specify allowed login domains from rCtx.Authenticated.Organization.AllowedLoginDomains
```

Comment:
Waiting to add the allowed domains until #3657 is in, or this could result in incorrect login failures